src/features/view-subdomains/SubdomainTableCard/SubdomainTableCard.tsx

```diff
@@ -14,70 +12,60 @@ import { NFT } from '@zero-tech/zui/components/GridCard/templates/NFT';
 import styles from './SubdomainTableCard.module.scss';
 
 type SubdomainTableCardProps = {
-	domainId: string;
-	domainName: string;
-	domainMetadataUri: string;
-	paymentTokenData: TokenPriceInfo;
-	onCardClick: (e?: any, domainName?: string) => void;
+	zna: string;
+	onClick: (e?: any, domainName?: string) => void;
 };
 
 export const SubdomainTableCard: FC<SubdomainTableCardProps> = ({
-	domainId,
-	domainName,
-	domainMetadataUri,
-	onCardClick,
+	zna,
+	onClick,
 }) => {
 	const {
-		metrics,
+		highestBid,
 		buyNowPrice,
 		metadata,
-		isMetadataLoading,
-		isMetricsLoading,
-		imageAlt,
-		imageSrc,
+		image,
+		alt,
+		isLoadingMetrics,
+		isLoadingMetadata,
 		paymentTokenLabel,
-	} = useSubdomainData({
-		id: domainId,
-		zna: domainName,
-		metadataUri: domainMetadataUri,
+	} = useSubdomainTableItem({
+		zna,
 	});
 
-	const highestBidString = metrics?.highestBid
-		? formatEthers(metrics?.highestBid)
-		: undefined;
-
-	const buyNowPriceString = buyNowPrice ? formatEthers(buyNowPrice) : undefined;
-
-	const label =
-		(buyNowPriceString ? 'Buy Now' : 'Top Bid') + ' ' + paymentTokenLabel;
-
+	const buyNowPriceString = buyNowPrice ? buyNowPrice : highestBid;
+	const label = (buyNowPrice ? 'Buy Now' : 'Top Bid') + ' ' + paymentTokenLabel;
 	const button = buyNowPrice ? <BuyNowButton /> : <PlaceBidButton isRoot />;
 
+	const handleOnClick = useCallback(() => {
+		onClick(undefined, zna);
```

Comment:
I think we'll need to change `undefined` to an `event` to resolve the modal issue, but not required for this PR